kin-sdk-core/src/androidTest/java/kin/sdk/core/KinAccountTest.java

```diff
@@ -114,58 +108,48 @@ public void getPublicKey_deletedAccount()
-        }
+    @Test(expected = OperationFailedException.class)
+    public void sendTransactionSync_negativeAmount() throws Exception {
+        kinAccount.sendTransactionSync(TO_ADDRESS, PASSPHRASE, new BigDecimal(-1));
     }
 
-    @Test
-    public void sendTransactionSync_wrongPublicAddress() {
-        try {
-            kinAccount.sendTransactionSync(null, PASSPHRASE, new BigDecimal(0));
-        } catch (Exception e) {
-            assertTrue(e instanceof OperationFailedException);
-            assertEquals("Addressee not valid - public address can't be null or empty", e.getMessage());
-        }
+    @Test(expected = OperationFailedException.class)
+    public void sendTransactionSync_nullPublicAddress() throws Exception {
+        kinAccount.sendTransactionSync(null, PASSPHRASE, new BigDecimal(0));
+    }
 
-        try {
-            kinAccount.sendTransactionSync("0xShortAddress", PASSPHRASE, new BigDecimal(0));
-        } catch (Exception e) {
-            assertEquals("invalid address hex length: 12 != 40", e.getCause().getMessage());
-        }
+    @Test(expected = OperationFailedException.class)
+    public void sendTransactionSync_shortPublicAddress() throws Exception {
+        kinAccount.sendTransactionSync("0xShortAddress", PASSPHRASE, new BigDecimal(0));
+    }
 
-        try {
-            kinAccount.sendTransactionSync("", PASSPHRASE, new BigDecimal(0));
-        } catch (Exception e) {
-            assertTrue(e instanceof OperationFailedException);
-            assertEquals("Addressee not valid - public address can't be null or empty", e.getMessage());
-        }
+    @Test(expected = OperationFailedException.class)
+    public void sendTransactionSync_longPublicAddress() throws Exception {
+        kinAccount
+            .sendTransactionSync("0xLongAddressVeryLongMoreThan40CharsYouCanCount!!", PASSPHRASE, new BigDecimal(0));
     }
 
-    @Test
-    public void sendTransactionSync_wrongPassphrase() {
-        try {
-            kinAccount.sendTransactionSync(TO_ADDRESS, "wongPassphrase", new BigDecimal(0));
-        } catch (Exception e) {
-            assertTrue(e instanceof PassphraseException);
-        }
+    @Test(expected = OperationFailedException.class)
+    public void sendTransactionSync_illegalCharPublicAddress() throws Exception {
+        kinAccount.sendTransactionSync("0xabababababababababababababababababababaX", PASSPHRASE, new BigDecimal(0));
     }
 
-    @Test
+    @Test(expected = OperationFailedException.class)
+    public void sendTransactionSync_emptyPublicAddress() throws Exception {
+        kinAccount.sendTransactionSync("", PASSPHRASE, new BigDecimal(0));
+    }
+
+    @Test(expected = PassphraseException.class)
+    public void sendTransactionSync_wrongPassphrase() throws Exception {
+        kinAccount.sendTransactionSync(TO_ADDRESS, "wongPassphrase", new BigDecimal(0));
+    }
+
+    @Test(expected = PassphraseException.class)
```

Comment:
same here, only second sendTransaction should fail.

Reply:
fixed